### DISCUSSION TITLE
ESP8266 + tasmota basic + home assistant + push button

### DISCUSSION THREAD

**Discussion by lecocqe** [arendst/Tasmota#13872] [Q&A] [answered]
Hello,

i have a question. i flashed an ESP8266 with tasmota basic and used "generic (18)" template.
i have a led on D1 and push button on D3. pushing button toggles led, this is ok. mqtt settings are ok.
Home assistant detected device => ok.

now when button is toggled i want to see it as a switch in HA, i read the different manuals and i tried different options and until now i could not make it work... 

What am i missing ?
thank you for your help.

Eric

**Comment by lecocqe** [comment, marked as answer]
i found a solution: declaring a fake relay on an other pin.

**Reply by ascillato** [reply]
Check the docs. There are many setoptions to achieve what you want. Also, please do a quick search in issues. This has been already addressed many times and you will find plenty of examples and options to achieve that.
